base/uk.ac.stfc.isis.ibex.devicescreens/src/uk/ac/stfc/isis/ibex/devicescreens/DeviceScreenVariables.java

```diff
@@ -0,0 +1,144 @@
+    private static final String GET_SCREENS_SUFFIX = "GET_SCREENS";
+    private static final String SET_SCREENS_SUFFIX = "SET_SCREENS";
+    private static final String SCHEMA_SUFFIX = "SCREENS_SCHEMA";
+
+    private final ObservableFactory switchingObservableFactory;
+    private final WritableFactory switchingWritableFactory;
+
+    private final ForwardingObservable<DeviceScreensDescription> deviceScreensObservable;
+    private final Writable<DeviceScreensDescription> deviceScreensWritable;
+    private final ForwardingObservable<String> deviceScreenSchema;
+
+    private final String pvPrefix;
+
+    /**
+     * Default constructor.
+     */
+    public DeviceScreenVariables() {
+        this(new ObservableFactory(OnInstrumentSwitch.SWITCH), new WritableFactory(OnInstrumentSwitch.SWITCH), null);
+    }
+
+
+    /**
+     * Constructor used for testing, so we can mock out the factories.
+     *
+     * @param switchingObservableFactory an observable factory to be used, could
+     *            be a mock
+     * @param switchingWritableFactory a writable factory to be used, could be a
+     *            mock
+     * @param pvPrefix the PV prefix
+     */
+    public DeviceScreenVariables(ObservableFactory switchingObservableFactory, WritableFactory switchingWritableFactory,
+            String pvPrefix) {
+        this.switchingObservableFactory = switchingObservableFactory;
+        this.switchingWritableFactory = switchingWritableFactory;
+        this.pvPrefix = pvPrefix;
+        
+        deviceScreensObservable =
+                InstrumentVariables.convert(readCompressed(getPvPrefix() + BLOCKSERVER_ADDRESS + GET_SCREENS_SUFFIX),
+                        new DeviceScreensDescriptionXmlParser());
+        
+        deviceScreensWritable = convert(writeCompressed(getPvPrefix() + BLOCKSERVER_ADDRESS + SET_SCREENS_SUFFIX),
+                new DeviceScreenDescriptionToXmlConverter());
+
+        deviceScreenSchema = readCompressed(getPvPrefix() + BLOCKSERVER_ADDRESS + SCHEMA_SUFFIX);
+    }
+
+    /**
+     * Gets the observable for the get device screens PV.
+     * 
+     * @return an observable pointing at the get device screens PV
+     */
+    public ForwardingObservable<DeviceScreensDescription> getDeviceScreens() {
+        return deviceScreensObservable;
+    }
+
+    /**
+     * Gets the writable to the set device screens PV.
+     * 
+     * @return a writable pointing to the set device screens PV
+     */
+    public Writable<DeviceScreensDescription> getDeviceScreensSetter() {
+        return deviceScreensWritable;
+    }
+
+    /**
+     * Gets the observable for the PV containing the device screens schema.
+     * 
+     * @return an observable pointing at the device screens schema PV
+     */
+    public ForwardingObservable<String> getDeviceScreensSchema() {
+        return deviceScreenSchema;
+    }
+
+    private ForwardingObservable<String> readCompressed(String address) {
```

Comment:
This is identical code to Config server line 205 and similar to Varibales in synoptic internal line 180. Is this intentional to have multiple copies?